libs/execution_context/ebpf_maps.c

```diff
@@ -1071,15 +1074,14 @@ _update_hash_map_entry_with_handle(
     uint8_t* old_value = NULL;
     ebpf_result_t found_result = ebpf_hash_table_find((ebpf_hash_table_t*)map->data, key, &old_value);
     ebpf_id_t old_id = (old_value) ? *(ebpf_id_t*)old_value : 0;
-    ebpf_core_object_t* value_object = NULL;
 
     if ((entry_count == map->ebpf_map_definition.max_entries) && (found_result != EBPF_SUCCESS)) {
         // The hash table is already full.
         result = EBPF_OUT_OF_SPACE;
         goto Done;
     }
 
-    result = _get_map_value_object(map, value_handle, value_type, &value_object);
+    result = _validate_map_value_object(object_map, value_type, value_object);
     if (result != EBPF_SUCCESS) {
         goto Done;
     }
```

Comment:
there's a call to `ebpf_object_release_reference` within lock. Is that OK? In `_update_array_map_entry_with_handle` you changed to call it outside lock.

Reply:
It should not make any difference. But still reverted to the original behavior (releasing reference under lock).